autotx/agents/SendTokensAgent.py

```diff
@@ -13,109 +13,63 @@
 
 system_message = lambda autotx: dedent(f"""
     You are an AI assistant. Assist the user (address: {autotx.manager.address}) in their tasks by fetching balances and preparing transactions to send tokens.
-    You are an expert in Ethereum tokens and can help users send tokens and check their balances.
+    You are an expert in Ethereum tokens (native and erc20) and can help users send tokens and check their balances.
     You use the tools available to assist the user in their tasks. 
     Your job is to only prepare the transactions and the user will take care of executing them.
-    NOTE: There is no reason to call get_erc20_balance after calling transfer as the transfers are only prepared and not executed. 
+    NOTE: There is no reason to call get_balance after calling transfer as the transfers are only prepared and not executed. 
     """
 )
 
-class TransferETHTool(AutoTxTool):
-    name: str = "prepare_transfer_eth_transaction"
+class TransferTokenTool(AutoTxTool):
+    name: str = "prepare_transfer_transaction"
     description: str = dedent(
         """
-        Prepares a transfer transaction for given amount in decimals for ETH.
+        Prepares a transfer transaction for given amount in decimals for a token
         """
     )
 
-    def build_tool(self, autotx: AutoTx) -> Callable[[float, str], str]:
+    def build_tool(self, autotx: AutoTx) -> Callable[[float, str, str], str]:
         def run(
             amount: Annotated[float, "Amount given by the user to transfer. The function will take care of converting the amount to needed decimals."],
-            receiver: Annotated[str, "The receiver's address or ENS domain"]
+            receiver: Annotated[str, "The receiver's address or ENS domain"],
+            token: Annotated[str, "Symbol of token to transfer"]
         ) -> str:
             web3 = load_w3()
-      
             receiver_addr = ETHAddress(receiver, web3)
-        
-            tx = build_transfer_eth(web3, ETHAddress(ADDRESS_ZERO, web3), receiver_addr, amount)
-        
-            prepared_tx = PreparedTx(f"Transfer {amount} ETH to {str(receiver_addr)}", tx)
-
-            autotx.transactions.append(prepared_tx)
+            token_address = ETHAddress(autotx.network.tokens[token.lower()], web3)
 
-            print(f"Prepared transaction: {prepared_tx.summary}")
-            
-            return prepared_tx.summary
+            prepared_tx: PreparedTx | None = None
 
+            if token_address.hex == NATIVE_TOKEN_ADDRESS:
+                tx = build_transfer_eth(web3, ETHAddress(ADDRESS_ZERO, web3), receiver_addr, amount)
 
-        return run
-        
-class TransferERC20Tool(AutoTxTool):
-    name: str = "prepare_transfer_erc20_transaction"
-    description: str = dedent(
-        """
-        Prepares a transfer transaction for given amount in decimals for given token.
-        """
-    )
+                prepared_tx = PreparedTx(f"Transfer {amount} ETH to {str(receiver_addr)}", tx)
```

Comment:
Same point as previous comment, this might be `MATIC` - We can use the token symbol here, right? Which makes me think that probably we can take the `prepared_tx` outside of this if condition